apps/homescreen/js/app.js

```diff
@@ -286,7 +295,30 @@ const SETTINGS_VERSION = 0;
         console.error('Failed to initialise metadata db', e);
         resolve();
       });
-    }).then(populateApps);
+    }).then(() => {
+      if (this.firstRun) {
+        return LazyLoader.load(['js/firstrun.js'],
+          () => {
+            FirstRun().then((results) => {
+              this.small = results.small;
+              this.icons.classList.toggle('small', this.small);
+              this.saveSettings();
+
+              this.startupMetadata = results.order;
+              populateApps();
+            }, (e) => {
+              console.error('Error running first-run script', e);
+              populateApps();
+            });
+          },
+          (e) => {
+            console.error('Failed to load first-run script');
+            populateApps();
+          });
+      }
+
+      return populateApps();
```

Comment:
Just that populateApps is a promise, so if we ever wanted to chain in the future, we wouldn't spend ages debugging why it seems to execute immediately (something I ran into elsewhere :))